R/shared-methods.R

```diff
@@ -509,6 +513,65 @@ DataClass <- R6::R6Class(
         source_data_cols = paste(unlist(self$source_data_cols), collapse = ", ")
       )
       return(sum_df)
+    },
+
+    #' @description Run tests on a country class instance. Calling `test()` on a
+    #' class instance runs tests with the settings in use. For example, if you
+    #' set `level = "1"` and `localise = FALSE` the tests will be run on level 1
+    #' data which is not localised. Rather than downloading data for a test
+    #' users can provide a path to a snapshot file of data to test instead.
+    #' Tests are run on a clone of the class. This method calls generic tests
+    #' for all country class objects. It also calls country specific tests
+    #' which can be defined in an individual country class method called
+    #' `specific_tests()`. The snapshots contain the first 1000 rows of data.
+    #' For more details see the 'testing' vignette.
```

Comment:
mention that: can build the vignette using vignette(vignette name) + link to our hosted site version.